framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -249,6 +249,15 @@
      */
     private final Map<String, AnnotationMirror> aliases = new HashMap<String, AnnotationMirror>();
 
+    /** Set contains class names of aliases whose elements need to be copied over. */
+    private final Set<String> aliasesCopyElements = new HashSet<>();
+
+    /**
+     * Map from class name (canonical name) of an annotation, to the ignorable elements that the
+     * framework can safely drop when copying over the elements.
+     */
+    private final Map<String, String[]> aliasesIgnorableElements = new HashMap<>();
+
     /**
      * A map from the class name (canonical name) of an annotation to the set of class names
      * (canonical names) for annotations with the same meaning (i.e., aliases), as well as the
```

Comment:
If you also want to fix an existing inconsistency, I would remove the second `canonical names` here, as it is an alias name, not a canonical name.

Reply:
I think the `canonical` here means `#Class.getCanonicalName`. To avoid confusion I will remove it anyway.